### PR TITLE
feat: remove app timeout

### PR DESCRIPTION
**Pull request by yanksyoon**
#### Description

*\<Please add why this change is needed and what it does.\>*
Adds timeout parameter to `model.remove_application()`
Addresses #855 

*\<Fixes: \>*

#### QA Steps

*\<Commands / tests / steps to run to verify that the change works:\>*
```
tox -e integration -- -k test_app_remove_timeout
```

All CI tests need to pass.

*\<Please note that most likely an additional test will be required by the reviewers for any change that's not a one liner to land.\>*

#### Notes & Discussion

*\<Additional notes for the reviewers if needed. Please delete section if not applicable.\>*

### CODE REVIEW

**Comment by tlm** [comment]
@cderici same again :)

File: tests/integration/test_application.py [hunk 1]
```diff
@@ -324,6 +325,16 @@ async def test_app_remove_wait_flag():
         assert a_name not in model.applications
 
 
+@base.bootstrapped
+async def test_app_remove_timeout():
+    async with base.CleanModel() as model:
+        app = await model.deploy('ubuntu')
```
Comment (cderici) [inline tests/integration/test_application.py:331]:
```suggestion
        app = await model.deploy('juju-qa-test')
```

**Review by cderici — APPROVED** [review]
This LGTM. Thanks for the change ❤️The test might fail in the CI (because we're not doing the best job of handling exceptions in the asyncio event loop - and for some reason the GH actions exposes that), however, it passes for me locally. One little change in the test and we can land this safely.

**Pull request merged**